### PR TITLE
4837 - Font awesome icons not loading in the administration console

### PR DESCRIPTION
**Pull request by kennsippell**
# Description

Font awesome icons not loading in the administration console. Relative paths to FontAwesome rely on a trailing `/` at the end of the url.  `/medic/_design/medic-admin/_rewrite/`

medic/medic-webapp#4837

This relative pathing seems a tad brittle.

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

**Comment by garethbowen** [comment]
I think we need to be able to work whether there's a trailing slash or not - how does webapp include font-awesome? Can we copy that?

**Comment by kennsippell** [comment]
So here is what's up. Both apps handle two different paths for the same font-awesome resources depending on the trailing slash.

App | Trailing Slash | FontAwesome Url
------------ | ------------- | -------------
WebApp | Yes | http://localhost:5988/medic/_design/medic/_rewrite/fonts/fontawesome-webfont.woff2
WebApp | No | http://localhost:5988/medic/_design/medic/fonts/fontawesome-webfont.woff2
Admin | Yes | http://localhost:5988/medic/_design/medic-admin/fonts/fontawesome-webfont.woff2
Admin | No | http://localhost:5988/medic/_design/fonts/fontawesome-webfont.ttf

Admin paths are "one directory up" from comparable scenario in WebApp.
Admin with no trailing slash results in 404. All others succeed.

The reason this is happening is in `webapp/node_modules/font-awesome/less/variables.less` which defines all font-awesome paths as a relative folder `@fa-font-path: "../fonts"`. This gets compiled into both `medic/_design/medic/_rewrite/css/inbox.css` and `medic/_design/medic-admin/_rewrite/main.css`.  Relative paths in CSS are relative to the location of the CSS file. So the decision to put inbox.css into `css/inbox.css` is why WebApp doesn't need the trailing slash for font-awesome.

So potential solutions:
1. Put admin's `main.css` into `css/main.css` and add a rewrite into webapp-admin's ddoc.
1. Use the grunt *patch* mechanism, to set `@fa-font-path: "./fonts"` (one dot instead of two) for `admin/node_modules`. Or both admin and webapp, but then I'll have to move main.css out of the css folder.

Don't love either solution. Inclined to go with 1 since the build-time *patch* stuff isn't a super obvious flow, particularly when different apps with the same resources require different patches to function.

File: webapp/src/js/services/location.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ angular.module('inboxServices').factory('Location',
     var location = $window.location;
     var dbName = location.pathname.split('/')[1];
     var path = '/' + dbName + '/_design/medic/_rewrite';
-    var adminPath = '/' + dbName + '/_design/medic-admin/_rewrite';
+    var adminPath = '/' + dbName + '/_design/medic-admin/_rewrite/';
```
Comment (dianabarsan) [inline webapp/src/js/services/location.js]: I think this change is no longer necessary.

File: ddocs/medic-admin/rewrites.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 [
   {"from": "/",             "to": "index.html"},
-  {"from": "/main.css",     "to": "main.css"},
+  {"from": "/css/*",        "to": "css/*"},
+  {"from": "/fonts/*",      "to": "fonts/*"},
```
Comment (dianabarsan) [inline ddocs/medic-admin/rewrites.json]: I think we should be consistent about all attachments and prefix the others well - just like Webapp.

Comment (kennsippell) [reply ddocs/medic-admin/rewrites.json]: Ya that makes sense. Looks like only /js/ resources in admin, so I moved those.

**Comment by kennsippell** [comment]
Reliability failure? Requesting re-kick?

```Failures:
1) Login tests :  should try to sign in and verify that credentials were incorrect
  Message:
    Failed: Wait timed out after 10002ms
  Stack:
    TimeoutError: Wait timed out after 10002ms
        at /home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2201:17
        at ManagedPromise.invokeCallback_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:1376:14)
        at TaskQueue.execute_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:3084:14)
        at TaskQueue.executeNext_ (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:3067:27)
        at asyncRun (/home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:2927:27)
        at /home/travis/build/medic/medic-webapp/node_modules/selenium-webdriver/lib/promise.js:668:7
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:189:7)```

**Pull request merged**